### PR TITLE
Convert gradle build scripts to Kotlin: part 1

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: custom/build.gradle.kts [hunk 1]
```diff
@@ -28,20 +28,29 @@ dependencies {
...
 tasks {
   compileJava {
     options.release.set(8)
   }
 
-  processResources {
-    expand([
-        'version': project.version
-    ])
+  register("generateVersionResource") {
```
Comment (iNikem) [inline custom/build.gradle.kts:48]: Why this change? Kotlin script does not support `processResources` task?

Comment (mateuszrzeszutek) [reply custom/build.gradle.kts:48]: Honestly, I don't know - it passed locally on my machine, it failed in GHA:

```
Execution failed for task ':custom:processResources'.
> Could not copy file '/home/runner/work/splunk-otel-java/splunk-otel-java/custom/src/main/resources/splunk.properties' to '/home/runner/work/splunk-otel-java/splunk-otel-java/custom/build/resources/main/splunk.properties'.
   > java.lang.UnsupportedOperationException (no error message)
```

I couldn't reproduce it so I just looked up how OTel SDK does it and copied the solution.

Comment (iNikem) [reply custom/build.gradle.kts:48]: Can you add some TODO here? This is certainly a hack

**Review by iNikem — APPROVED** [review]

**Review by iNikem — APPROVED** [review]
